src/sentry/models/organization.py

```diff
@@ -252,12 +252,17 @@ def save(self, *args, **kwargs):
                 slugify_target = slugify_target.lower().replace("_", "-").strip("-")
                 slugify_instance(self, slugify_target, reserved=RESERVED_ORGANIZATION_SLUGS)
 
-        if SENTRY_USE_SNOWFLAKE:
-            self.save_with_snowflake_id(
-                self.snowflake_redis_key, lambda: super(Organization, self).save(*args, **kwargs)
-            )
-        else:
-            super().save(*args, **kwargs)
+        # Run the save + outbox queueing in a transaction to ensure the control-silo is notified
+        # when a change is made to the organization model.
+        with transaction.atomic():
+            if SENTRY_USE_SNOWFLAKE:
+                self.save_with_snowflake_id(
+                    self.snowflake_redis_key,
+                    lambda: super(Organization, self).save(*args, **kwargs),
```

Comment:
Ahhhh, so this logic has to get silly stupid.  It should be:

```
...
def save_with_outbox(*args, **kwargs):
   super().save(*args, **kwargs)
   Organization.outbox_for_update(self.id).save()

def save():
  if SENTRY_USE_SNOWFLAKE:
    self.save_with_snowflake_id(
                    self.snowflake_redis_key,
                    lambda: self.save_with_outbox(*args, **kwds))
  else:
    with transaction.atomic():
     self.save_with_outbox(*args, **kwds
```

The reason is that the callback here in the snowflake is wrapped inside an inner transaction that we should be part of.